### PR TITLE
Fix a bug when checking a pose is empty and TOTG corner case

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Fix: https://github.com/ros-planning/moveit_task_constructor/issues/350

The first commit fix `isEmpty` for ROS2 since `w == 1` 
The second commit fix a corner case for TOTG where it might not use the last input waypoint when parameterizing the trajectory

To test run:

```
ros2 launch moveit_task_constructor_demo demo.launch.py
ros2 launch moveit_task_constructor_demo pickplace.launch.py
```

### CODE REVIEW

File: moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp [hunk 1]
```diff
@@ -1096,6 +1096,10 @@ bool TimeOptimalTrajectoryGeneration::doTimeParameterizationCalculations(robot_t
 
     if (diverse_point)
       points.push_back(new_point);
+    // If the last point is not a diverse_point we replace the last added point with it to make sure to always have the
```
Comment (jliukkonen) [inline moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp:1099]: I don't have anything against the fix. But why is the variable called `diverse_point` when it seems to mean, based on the comments, simply "unique waypoint", which would be easier to understand?

File: moveit_core/utils/src/message_checks.cpp [hunk 2]
```diff
@@ -71,7 +71,7 @@ bool isEmpty(const moveit_msgs::msg::Constraints& constr)
 
 bool isEmpty(const geometry_msgs::msg::Quaternion& msg)
 {
-  return msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 0.0;
+  return msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 1.0;
```
Comment (AndyZe) [inline moveit_core/utils/src/message_checks.cpp:74]: I think this should cover both cases. Whether `w` is initialized or not initialized.

```suggestion
  return (msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 1.0) || (msg.x == 0.0 && msg.y == 0.0 && msg.z == 0.0 && msg.w == 0.0);
```

File: moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp (hunk 1 above)
Comment (JafarAbdi) [reply moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp:1099]: > I don't really understand the fix. If it's not a diverse_point, that means it's identical to the previous point, right?

It's identical with some tolerance `min_angle_change_`, it was causing [MTC to fail](https://github.com/ros-planning/moveit_task_constructor/blob/master/core/src/stages/connect.cpp#L127-L132) because the last waypoint after parameterizing the trajectory isn't the expected goal waypoint

File: moveit_core/utils/src/message_checks.cpp (hunk 2 above)
Comment (JafarAbdi) [reply moveit_core/utils/src/message_checks.cpp:74]: I don't think we should handle the other case since it no longer exist in ROS2

File: moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp (hunk 1 above)
Comment (AndyZe) [reply moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp:1099]: OK, that makes sense

**Review by AndyZe — APPROVED** [review]
